packages/did-core-test-server/suites/did-consumption/did-json-consumption.js

```diff
@@ -0,0 +1,133 @@
+const deepEqual = require('deep-equal')
+const {isXmlDatetime} = require('../utils');
+
+const generateJsonConsumptionTests = (
+  {did, didDocumentDataModel, resolutionResult}) => {
+  const {representation} = resolutionResult;
+  const didDocument = JSON.parse(representation);
+
+  describe('6.2.2 JSON Consumption - The DID document and DID document ' +
+    'data structures JSON representation MUST be deserialized into the ' +
+    'data model according to the following consumption rules:', () => {
+    allValues = _getAllValues(didDocument);
+
+    allValues.forEach(value => {
+      if(Array.isArray(value)) {
+        it('6.2.2 JSON Consumption - JSON Array where the data model entry ' +
+          'value is a list or unknown: A list, where each value of the JSON ' +
+          'Array is added to the list in order, converted based on the JSON ' +
+          'representation type of the array value, as defined in this ' +
+          'table.', () => {
+          expect(reserializationSuccess(value)).toBe(true);
+        });
+        it('6.2.2 JSON Consumption - JSON Array where the data model entry ' +
+          'value is a set: A set, where each value of the JSON Array is ' +
+          'added to the set in order, converted based on the JSON ' +
+          'representation type of the array value, as defined in this ' +
+          'table.', () => {
+          expect(reserializationSuccess(value)).toBe(true);
+        });
+      } else if(value !== null && typeof value === 'object') {
+        it('6.2.2 JSON Consumption - JSON Object: A map, where each member ' +
+          'of the JSON Object is added as an entry to the map. Each entry ' +
+          'key is set as the JSON Object member name. Each entry value is ' +
+          'set by converting the JSON Object member value according to the ' +
+          'JSON representation type as defined in this table. Since order ' +
+          'is not specified by JSON Objects, no insertion order is ' +
+          'guaranteed.', () => {
+          expect(reserializationSuccess(value)).toBe(true);
+        });
+      } else if(typeof value === 'string' && isXmlDatetime(value)) {
+        it('6.2.2 JSON Consumption - JSON String where data model entry ' +
+          'value is a datetime: A datetime.', () => {
+          expect(reserializationSuccess(value)).toBe(true);
```

Comment:
This test does not test whether the value is valid DateTime string or not.

Reply:
It does. It has to be a valid XML Datetime for the test to fire. The consumption test then checks for round-tripping. I do agree that it's not a straightforward test, but the check is being performed (even though it's not obvious it is).